raiden/tests/unit/test_channelstate.py

```diff
@@ -123,6 +133,43 @@ def create_channel_from_models(our_model, partner_model):
         )
     )
 
+    our_nonce = our_model.next_nonce - 1
+    assert our_nonce >= 0, "nonce cannot be negative"
+    if our_nonce > 0:
+        our_unsigned = create(
+            BalanceProofProperties(
+                nonce=our_nonce,
+                transferred_amount=0,
+                locked_amount=len(our_model.merkletree_leaves),
+                locksroot=merkleroot(channel_state.our_state.merkletree),
+                canonical_identifier=channel_state.canonical_identifier,
+            )
+        )
+    else:
+        our_unsigned = None
+
+    partner_nonce = partner_model.next_nonce - 1
+    assert partner_nonce >= 0, "nonce cannot be negative"
+    if partner_nonce > 0:
+        partner_unsigned = create(
+            BalanceProofProperties(
+                nonce=partner_nonce,
+                transferred_amount=0,
+                locked_amount=len(partner_model.merkletree_leaves),
+                locksroot=merkleroot(channel_state.partner_state.merkletree),
+                canonical_identifier=channel_state.canonical_identifier,
+            )
+        )
+
+        partner_signed = make_signed_balance_proof_from_unsigned(
+            partner_unsigned, LocalSigner(partner_pkey)
+        )
+    else:
+        partner_signed = None
+
+    channel_state.our_state.balance_proof = our_unsigned
+    channel_state.partner_state.balance_proof = partner_signed
```

Comment:
The factory was missing the balance proofs. I had to add this because clearing the channel state depends on the locksroot, and the value for the BP is used.